plugin/rewrite/name.go

```diff
@@ -16,6 +18,36 @@ func newNameRule(from, to string) (Rule, error) {
 
 // Rewrite rewrites the the current request.
 func (rule *nameRule) Rewrite(w dns.ResponseWriter, r *dns.Msg) Result {
+	rname := r.Question[0].Name
+	// Rewrite based on substring
+	if strings.Contains(rule.From, "sub:") {
+		rwsub := strings.Replace(rule.From, "sub:", "", -1)
+		if strings.Contains(rname, rwsub) {
+			r.Question[0].Name = strings.Replace(rname, rwsub, rule.To, -1)
+			return RewriteDone
+		}
+	}
+	// Rewrite based on regular expression
+	if strings.Contains(rule.From, "regex:") {
+		pattern := strings.Replace(rule.From, "regex:", "", -1)
+		regexPattern, err := regexp.Compile(pattern)
+		if err != nil {
+			return RewriteIgnored
+		}
+		regexGroups := regexPattern.FindStringSubmatch(rname)
+		if len(regexGroups) == 0 {
+			return RewriteIgnored
+		}
+		for i, v := range regexGroups {
+			rwi := fmt.Sprintf("{%d}", i)
```

Comment:
`strconv.Itoa` will be slightly cheaper to convert the int to a string.

Reply:
> strconv.Itoa will be slightly cheaper to convert the int to a string.

Agree 👍 will re-work